components/automate-ui/src/app/modules/infra-proxy/chef-servers-list/chef-servers-list.component.ts

```diff
@@ -0,0 +1,135 @@
+import { Component, OnInit, OnDestroy, EventEmitter } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Store, select } from '@ngrx/store';
+import { filter, takeUntil, map } from 'rxjs/operators';
+import { Regex } from 'app/helpers/auth/regex';
+import { Observable, Subject, combineLatest } from 'rxjs';
+import { isNil } from 'lodash/fp';
+import { HttpStatus } from 'app/types/types';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { LayoutFacadeService } from 'app/entities/layout/layout.facade';
+import { loading, EntityStatus, pending } from 'app/entities/entities';
+import { Server } from 'app/entities/servers/server.model';
+import {
+  allServers, getStatus, saveStatus, saveError } from 'app/entities/servers/server.selectors';
+import { CreateServer, GetServers, DeleteServer } from 'app/entities/servers/server.actions';
+import { ChefSorters } from 'app/helpers/auth/sorter';
+
+@Component({
+  selector: 'app-chef-servers-list',
+  templateUrl: './chef-servers-list.component.html',
+  styleUrls: ['./chef-servers-list.component.scss']
+})
+export class ChefServersListComponent implements OnInit, OnDestroy {
+  public loading$: Observable<boolean>;
+  public sortedChefServers$: Observable<Server[]>;
+  public createModalVisible = false;
+  public createChefServerForm: FormGroup;
+  public creatingChefServer = false;
+  public conflictErrorEvent = new EventEmitter<boolean>();
+  private isDestroyed = new Subject<boolean>();
+  public serverToDelete: Server;
+  public deleteModalVisible = false;
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    private fb: FormBuilder,
+    private layoutFacade: LayoutFacadeService
+  ) {
+    this.loading$ = store.pipe(select(getStatus), map(loading));
+
+    this.sortedChefServers$ = store.pipe(
+      select(allServers),
+      map(servers => ChefSorters.naturalSort(servers, 'name')));
+
+    this.createChefServerForm = this.fb.group({
```

Comment:
Please add a comment after this line like this:
https://github.com/chef/automate/blob/fd54bb3f252d715e44a877daab48c52ffb554e4e/components/automate-ui/src/app/modules/token/token-list/api-token-list.component.ts#L72-L73

... using your file name, create-chef-server-modal.component.html